internal/db/schema/migrations/oss/postgres/29/04_kms_keys.up.sql

```diff
@@ -0,0 +1,213 @@
+begin;
+
+-- make the required schema changes to adopt
+-- github.com/hashicorp/go-kms-wrapping/extras/kms/v2 
+-- this migration is from:
+-- https://github.com/hashicorp/go-kms-wrapping/blob/main/extras/kms/migrations/postgres/04_keys.up.sql 
+
+-- we can use our existing kms_root_key table.
+
+-- we need to add an add additional constraint to the kms_root_key_version
+-- table: 
+-- https://github.com/hashicorp/go-kms-wrapping/blob/c06f9db9380f2c26cad07cf51f7a324d25dd55ba/extras/kms/migrations/postgres/04_keys.up.sql#L29 
+alter table kms_root_key_version
+  add constraint not_empty_key
+  check (
+      length(key) > 0
+    );
+
+
+create table kms_data_key (
+  private_id kms_private_id primary key,
+  root_key_id kms_private_id not null
+    references kms_root_key(private_id)
+    on delete cascade
+    on update cascade,
+  purpose text not null
+    constraint not_start_end_whitespace_purpose
+    check (length(trim(purpose)) = length(purpose)),
+  create_time kms_timestamp,
+  unique (root_key_id, purpose) -- there can only be one dek for a specific purpose per root key
```

Comment:
Can you name this unique constraint with a name that follows our standard naming convention for unique constraints? Our unique constraint naming convention is to concatenate the table name with each column in the unique constraint and then add the suffix `uq`. (ex: `table_name_col1_col2_colN_uq`)

**Example** shows a foreign key constraint and a unique constraint:

```sql
create table widget_shelf (
  public_id wt_public_id primary key,
  scope_id wt_scope_id not null
    constraint iam_scope_fkey
      references iam_scope (public_id)
      on delete cascade
      on update cascade,
  name wt_name,
  constraint widget_shelf_scope_id_name_uq
    unique(scope_id, name)
);
```